From dfecf4e7a5bf205cae9120a8c77ea0dcd72f39a9 Mon Sep 17 00:00:00 2001
From: Ravi Prakash <raviprak@altiscale.com>
Date: Mon, 26 Jun 2017 13:24:27 -0700
Subject: [PATCH 2785/2848] HDFS-11993. Add log info when connect to datanode
 socket address failed. Contributed by chencan

(cherry picked from commit a9d3412b4ce40f5ab5a18756ede7e0606b653171)
(cherry picked from commit ef491f463799bdfd757d563bee145ce52765d3de)

 Conflicts:
	hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java

Change-Id: I079be7296c04c03e50ba61ffd30e5955591de793
(cherry picked from commit 493462e4b7d4e9c94c329cfd0f592647087eca30)
---
 .../org/apache/hadoop/hdfs/DFSInputStream.java     |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java
index a861946..a03889a 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java
@@ -701,8 +701,9 @@ private synchronized DatanodeInfo blockSeekTo(long target) throws IOException {
           fetchBlockAt(target);
         } else {
           connectFailedOnce = true;
-          DFSClient.LOG.warn("Failed to connect to " + targetAddr + " for block"
-            + ", add to deadNodes and continue. " + ex, ex);
+          DFSClient.LOG.warn("Failed to connect to " + targetAddr
+              + " for block " + targetBlock.getBlock()
+              + ", add to deadNodes and continue. ", ex);
           // Put chosen node into dead list, continue
           addToDeadNodes(chosenNode);
         }
-- 
1.7.9.5

